internal/controller/isbservicerollout_controller.go

```diff
@@ -234,11 +234,17 @@ func processISBServiceStatus(ctx context.Context, isbsvc *kubernetes.GenericObje
 	isbSvcPhase := numaflowv1.ISBSvcPhase(isbsvcStatus.Phase)
 	switch isbSvcPhase {
 	case numaflowv1.ISBSvcPhaseFailed:
-		rollout.Status.MarkChildResourcesUnhealthy("ISBSvcFailed", "ISBService Failed")
+		rollout.Status.MarkChildResourcesUnhealthy("ISBSvcFailed", "ISBService Failed", rollout.Generation)
+	case numaflowv1.ISBSvcPhasePending:
+		rollout.Status.MarkChildResourcesUnhealthy("ISBSvcPending", "ISBService Pending", rollout.Generation)
 	case numaflowv1.ISBSvcPhaseUnknown:
-		// this will have been set to Unknown in the call to InitConditions()
+		rollout.Status.MarkChildResourcesHealthUnknown("ISBSvcUnknown", "ISBService Phase Unknown", rollout.Generation)
 	default:
-		rollout.Status.MarkChildResourcesHealthy()
+		if rollout.Generation == isbsvcStatus.ObservedGeneration {
+			rollout.Status.MarkChildResourcesHealthy(rollout.Generation)
+		} else {
+			rollout.Status.MarkChildResourcesUnhealthy("GenerationMismatch", "ISBService Generation mismatch the ObservedGeneration", rollout.Generation)
```

Comment:
it looks like in the NumaflowControllerRollout reconciler we call this "Progressing" but call it "GenerationMismatch" here?

Reply:
Good catch. I think we should use `Progressing` for all instead of `GenerationMismatch` unless we think that there could be a case in which the generation and observedGeneration will never match. What do you think?

Reply:
I think we should use Progressing as well. Keep it standard with what `Deployments` do.